src/FormTaxonomies/Actions/EnqueueFormBuilderAssets.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Give\FormTaxonomies\Actions;
+
+class EnqueueFormBuilderAssets
+{
+    /**
+     * @var array
+     */
+    protected $settings;
+
+    public function __construct()
+    {
+        $this->settings = give_get_settings();
+    }
+
+    public function __invoke()
+    {
+        if($this->isFormTagsEnabled() || $this->isFormCategoriesEnabled()) {
+            wp_enqueue_script('givewp-builder-taxonomy-settings', GIVE_PLUGIN_URL . 'assets/dist/js/form-taxonomy-settings.js');
+            wp_add_inline_script('givewp-builder-taxonomy-settings','var giveTaxonomySettings =' . json_encode([
+                    'formTagsEnabled' => $this->isFormTagsEnabled(),
+                    'formCategoriesEnabled' => $this->isFormCategoriesEnabled(),
+                    'formTagsSelected' => $this->getSelectedFormTags(),
+                    'formCategoriesAvailable' => $this->getFormCategories(),
+                    'formCategoriesSelected' => $this->getSelectedFormCategories(),
+                ]));
+        }
+    }
+
+    public function isFormTagsEnabled()
+    {
+        return give_is_setting_enabled($this->settings['tags']);
+    }
+
+    public function isFormCategoriesEnabled()
+    {
+        return give_is_setting_enabled($this->settings['categories']);
+    }
+
+    public function getSelectedFormTags(): array
+    {
+        $terms = get_terms([
+            // Form ID not provided by the hook, so we need to get it from the query string (if available).
+            'post' => absint($_GET['donationFormID'] ?? 0),
+            'taxonomy' => 'give_forms_tag',
+        ]);
+
+        return array_map(function ($term) {
+            return [
+                'id' => $term->term_id,
+                'value' => $term->name,
+            ];
+        }, $terms) ?? [];
+    }
+
+    public function getFormCategories(): array
```

Comment:
Can you please add doc blocks for all the methods here and include some unit tests?  I could see a view model being useful here with a method for exports that can be asserted

Reply:
I extracted the methods to a view model ([56ea12e](https://github.com/impress-org/givewp/pull/7463/commits/56ea12e65678fb7b6732e2f1c54816b936897801)) and added some tests ([04bab95](https://github.com/impress-org/givewp/pull/7463/commits/04bab95ff10db7fc8f139d895960811375101cbd)).